pokedex.go

```diff
@@ -1,14 +1,17 @@
 package pokemonbattlelib
 
+import "log"
+
 //go:generate go run data/gen.go
 
-// Get a new pokemon by nation dex number.
-func GetPokemon(natdex uint16) Pokemon {
+// Creates a new Pokemon given its national dex number
+func NewPokemon(natdex uint16) *Pokemon {
```

Comment:
This either needs to explicitly copy the pokemon, or should not return a pointer, because then the Pokemon being referenced is the one inside of `ALL_POKEMON`